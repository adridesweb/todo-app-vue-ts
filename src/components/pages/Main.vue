<template>
  <section class="main" id="main">
    <TodoList />
  </section>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import TodoList from "@/components/TodoList.vue";

export default defineComponent({
  name: "Main",
  components: {
    TodoList,
  },
});
</script>

<style scoped>
.main {
  position: relative;
  z-index: 2;
}
</style>
