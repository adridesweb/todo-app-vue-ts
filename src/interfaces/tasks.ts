export interface Task {
  id: string | number;
  title: string;
  completed: boolean;
  editing: boolean;
}
