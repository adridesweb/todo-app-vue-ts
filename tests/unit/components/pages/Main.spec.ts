import { mount } from "@vue/test-utils";
import { v4 as uuid } from "uuid";
import { Task } from "@/interfaces/tasks";

import Main from "@/components/pages/Main.vue";
import createVuexStore from "../../mock-data/mock-store";

describe("tests on pages/Main.vue", () => {
  const newBasicTask = <Task>{
    id: uuid(),
    title: "New task",
    completed: false,
    editing: false,
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly with one task in store", () => {
    const store = createVuexStore({ tasks: [newBasicTask] });
    const wrapper = mount(Main, {
      global: {
        plugins: [store],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
