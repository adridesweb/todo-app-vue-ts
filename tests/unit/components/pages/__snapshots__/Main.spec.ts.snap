// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on pages/Main.vue renders correctly with one task in store 1`] = `
<section
  class="main"
  id="main"
>
  <ul
    class="todo-list"
  >
    
    <li
      class=""
    >
      <div
        class="view"
      >
        <input
          class="toggle"
          type="checkbox"
        />
        <label>
          New task
        </label>
        <button
          class="destroy"
        />
      </div>
      <input
        class="edit"
      />
    </li>
    
  </ul>
</section>
`;
